### ISSUE TITLE
PostUp iptables masquerade directive may use interface of default route instead of hardcoded eth0

### ISSUE THREAD

**Issue by chrisbraucker** [linuxserver/docker-wireguard#169]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

------------------------------

## Expected Behavior

When more than one network is attached to the container, and the default route does not point to `eth0`, but e.g. `ethX`, PostUp/PostDown should possibly execute `iptables -t nat -A POSTROUTING -o ethX -j MASQUERADE` instead (in server.conf). Though I've run into testcases where the default route did not point to the interface with internet access, and then this would be moot. (See below)

## Current Behavior

`eth0` is hardcoded and has to be adjusted manually in generated `wg0.conf`. I'd like to file a PR but do not fully understand the structure of [the init](https://github.com/linuxserver/docker-wireguard/blob/master/root/etc/cont-init.d/40-confs).
I can, however, offer a PR with additional documentation for such scenarios.

Output of `ip route`:
```
default via 172.18.0.1 dev eth1
10.0.1.0/24 dev eth0 proto kernel scope link src 10.0.1.47
10.13.13.2 dev wg0 scope link
10.13.13.3 dev wg0 scope link
172.18.0.0/16 dev eth1 proto kernel scope link src 172.18.0.3
```

Output of `ip list`:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
3: wg0: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN group default qlen 1000
    link/none
    inet 10.13.13.1/32 scope global wg0
       valid_lft forever preferred_lft forever
137: eth0@if138: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1450 qdisc noqueue state UP group default
    link/ether 02:42:0a:00:01:2f brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.0.1.47/24 brd 10.0.1.255 scope global eth0
       valid_lft forever preferred_lft forever
139: eth1@if140: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:ac:12:00:03 brd ff:ff:ff:ff:ff:ff link-netnsid 1
    inet 172.18.0.3/16 brd 172.18.255.255 scope global eth1
       valid_lft forever preferred_lft forever
```

## Steps to Reproduce
1. Use the provided docker-compose.yml configuration (bump version to >3, run in server mode, e.g. PEERS=1) on a docker-swarm enabled host with `docker stack deploy` (single node suffices)
2. this usually creates a separate (overlay) network for the stack and also attaches the hosts' default bridge network for internet access
3. Witness `ip route` configuration not necessarily pointing the default route to `eth0` but `eth1`

Alternatively use the following docker-compose.yml
```
version: "3.8"
services:
  wireguard:
    image: lscr.io/linuxserver/wireguard:latest
    cap_add:
      - NET_ADMIN
      - SYS_MODULE
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Amsterdam
      - SERVERURL=auto
      - SERVERPORT=51820
      - PEERS=1
      - PEERDNS=8.8.8.8
      - INTERNAL_SUBNET=10.13.13.0/23
      - ALLOWEDIPS=0.0.0.0/0
      - LOG_CONFS=false
    ports:
      - 51820:51820/udp
    networks:
      - net1
      - net2
    sysctls:
      - net.ipv4.conf.all.src_valid_mark=1

networks:
  net1:
    driver: bridge
    internal: true
  net2:
    driver: bridge
```
This way, the default route may point to the right interface, but possibly not. Checking with `ping -I $IFACE google.com` shows that one interface does not have internet access.

Thank you for your time.
And if addressing this is beyond scope, I understand.

**Comment by aptalca** [comment]
We don't test or support swarm. But you can easily adjust the behavior by editing the server.conf template in the config folder and force conf regeneration:
https://github.com/linuxserver/docker-wireguard/blob/master/root/defaults/server.conf

**Comment by chrisbraucker** [comment]
Thank you for the reply.
The issue occurred to me with swarm, but it's not limited to that. That's why I also added a regular docker-compose.yml that showcases this issue outside of swarm.

Wrt. your suggestion, I adjusted the generated wg0.conf on the host file system and restarted the container. However, if more than one network is attached, it is not always straightforward where the routing will end up.
A hint in the README that this may happen (multiple networks interfering with the iptables rules) would have saved me quite some time, and possibly others, hence my offer to contribute docs. I'll file a PR with some docs for this tomorrow, as mitigation through code seems rather unreasonable in hindsight
